src/ROUTES.ts

```diff
@@ -226,6 +227,18 @@ const ROUTES = {
         route: 'r/:reportID/participants',
         getRoute: (reportID: string) => `r/${reportID}/participants` as const,
     },
+    REPORT_PARTICIPANTS_INVITE: {
+        route: 'r/:reportID/participants/invite',
+        getRoute: (reportID: string) => `r/${reportID}/participants/invite` as const,
+    },
+    REPORT_PARTICIPANTS_DETAILS: {
+        route: 'r/:reportID/participants/:accountID',
+        getRoute: (reportID: string, accountID: number) => `r/${reportID}/participants/${accountID}`,
```

Comment:
you need `as const` at the end of this string (and same for the next route) to fix the TS errors. Fortunately we have a good custom error message for this case:

```
Type 'string' does not satisfy the constraint '"Error: One or more routes defined within `ROUTES` have not correctly used `as const` in their `getRoute` function return value."'.
```

I suspect this may also be the cause of your lint errors

Reply:
confirmed, that change will fix your lint errors too

Reply:
Here are some [TS docs on const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions). In the case of strings, `as const` just means the type is literally that specific string, not just `string`:

```ts
// This variable's type will be "string"
let hello = 'hello';

// since it's any string, this works fine
hello = 'HELLO';

// This variable's type is 'world'
const world = 'world' as const;

// error - Type '"WORLD"' is not assignable to type '"world"'
world = 'WORLD';
```

![image](https://github.com/Expensify/App/assets/47436092/f193f226-9554-4bd8-a496-4eece4a61f0b)

Reply:
Thanks for the detailed explanation 🙇